CMakeLists.txt

```diff
@@ -132,7 +132,11 @@ if(GPU_ARCHS)
     unset(GPU_TARGETS CACHE)
     unset(AMDGPU_TARGETS CACHE)
 endif()
-
+if(GPU_TARGETS)
+    set(USER_GPU_TARGETS 1)
+else()
+    set(USER_GPU_TARGETS 0)
+endif()
 find_package(hip)
```

Comment:
@illsilin  LGTM, while it might be better to document the cause here, `GPU_TARGETS` is reset by `find_package(hip`) if not previously set, so it has to be above Line 140 here.